security/fma-contract-version-upgrade.md

```diff
@@ -0,0 +1,172 @@
+- **Detection:** Running unit tests as fork test.
+- **Recovery Path(s)**: Redeployment and upgrade of the affected contracts
+
+#### Chosen solidity version is too recent and might contain unknown bugs
+
+- **Description:** Choosing a very recent solidity version which hasn't stood a relatively long enough test of time can pose the risk of having an unknown bug introduced into the OP Stack smart contracts.
+- **Risk Assessment:**
+  - Severity: High
+  - Likelihood: Low/Medium
+- **Mitigations:** Might be safer to use a solidity version 1 or 2 releases earlier than the latest.
+- **Detection:** -
+- **Recovery Path(s)**: Redeployment and upgrade of the affected contracts
+
+#### Stack too deep errors during compilation
+
+- **Description:** Current OP Stack contracts only compile with the legacy pipeline with the optimizer on. Compiling with the optimizer off or using the via-ir pipeline result in stack too deep errors. When upgrading to a new solidity version, we have to make sure that successful compilation using the current build settings is possible.
+- **Risk Assessment:**
+  - Severity: -
+  - Likelihood: -
+- **Mitigations:** -
+- **Detection:** During compilation
+- **Recovery Path(s)**: Code modification (local variable scoping, local variables in memory etc), build settings modification or/and usage of a different compiler version.
+
+#### Contracts exceeding code size when compiled
+
+- **Description:** There are a few OP Stack contracts that are close to the contract code size limit. We have to ensure that the chosen compiler version to upgrade to compiles all OP Stack contracts successfully and without any exceeding the contract code size limit.
+- **Risk Assessment:**
+  - Severity: -
+  - Likelihood: -
+- **Mitigations:** -
+- **Detection:** During compilation
+- **Recovery Path(s)**: Code modification (runtime code size optimization), build settings modification or/and usage of a different compiler version.
+
+## Audit Requirements
+
+_Will this project require an audit according to the guidance in [OP Labs Audit Framework: When to get external security review and how to prepare for it](https://gov.optimism.io/t/op-labs-audit-framework-when-to-get-external-security-review-and-how-to-prepare-for-it/6864)? Please explain your reasoning._
+
+This project would not require an audit. An audit would require a bytecode audit which is not practical. Also, historically, we trust compiler generated code and don't audit bytecode so we are continuing to do the same here.
```

Comment:
small nit to clarify
```suggestion
This project would not require an audit. Because we don't expect to change Solidity code aside from the pragmas, an audit would have to review the bytecode diff, which is not practical. Also, historically, we trust compiler generated code and don't audit bytecode so we are continuing to do the same here.
```